lms/djangoapps/teams/static/teams/js/collections/topic.js

```diff
@@ -15,9 +15,15 @@
                     this.registerSortableField('name', gettext('name'));
                     // Translators: This refers to the number of teams (a count of how many teams there are)
                     this.registerSortableField('team_count', gettext('team count'));
+
+                    this.listenTo(Backbone, 'teamModel:add', this.refreshCurrentPage);
                 },
 
-                model: TopicModel
+                model: TopicModel,
+
+                refreshCurrentPage: function() {
+                    this.goTo(this.currentPage);
```

Comment:
I don't like the collection being responsible for views being refreshed. I'd rather that the views add listeners on the collection and then refresh themselves as needed.